pkgs/build-support/rust/build-rust-crate/configure-crate.nix

```diff
@@ -209,6 +210,9 @@ in ''
      EXTRA_LINK_LIBS=$(sed -n "s/^cargo::\{0,1\}rustc-link-lib=\(.*\)/\1/p" target/build/${crateName}.opt | tr '\n' ' ')
      EXTRA_LINK_SEARCH=$(sed -n "s/^cargo::\{0,1\}rustc-link-search=\(.*\)/\1/p" target/build/${crateName}.opt | tr '\n' ' ' | sort -u)
 
+     CRATE_TYPE_IS_CDYLIB="${if lib.elem "cdylib" crateType then "true" else ""}"
```

Comment:
Can you wrap these two in an inline optionalString?

Reply:
Thanks, good idea!
Fixed now